public/app/core/hooks/useMediaQueryChange.ts

```diff
@@ -1,17 +0,0 @@
-import { useEffect } from 'react';
-
-export function useMediaQueryChange({
-  breakpoint,
-  onChange,
-}: {
-  breakpoint: number;
-  onChange: (e: MediaQueryListEvent) => void;
-}) {
-  useEffect(() => {
-    const mediaQuery = window.matchMedia(`(min-width: ${breakpoint}px)`);
-    const onMediaQueryChange = (e: MediaQueryListEvent) => onChange(e);
-    mediaQuery.addEventListener('change', onMediaQueryChange);
-
-    return () => mediaQuery.removeEventListener('change', onMediaQueryChange);
-  }, [breakpoint, onChange]);
```

Comment:
because onChange was never memoized by any user of this hook this useEffect was always firing / cleanup on ever re-render. This caused (at least in explore which triggers a lot of AppChromeService state updates) bugs to appear. 

I think this hook was a bit poorly designed, better to return a "isMatch" state instead and have the change effect handled by the called in a proper useEffect.